lib/types.js

```diff
@@ -18,30 +19,87 @@ const exclude = function (types, excludes) {
     return types;
   }
 
-  for (const doc of types) {
+  const typesCopy = CloneDeep(types);
+
+  for (const doc of typesCopy) {
     for (const def of doc.definitions) {
       if (def.kind !== 'ObjectTypeDefinition') {
         continue;
       }
       if (['Query', 'Mutation', 'Subscription'].indexOf(def.name.value) > -1) {
         def.fields = def.fields.filter((field) => {
-          if (!check(def.name.value, field.name.value, excludes)) {
+          if (check(def.name.value, field.name.value, excludes)) {
+            debug(`excluding ${def.name.value}.${field.name.value} from import`);
+            return false;
+          }
+          return true;
+        });
+      }
+    }
+  }
+
+  return typesCopy;
+}
+
+const excludeImmutable = function (types, excludes) {
+  if (!excludes || excludes.length < 1) {
+    return types;
+  }
+
+  let mutations = [];
+
+  for (let typeIndex = 0; typeIndex < types.length; typeIndex++) {
+    const doc = types[typeIndex];
+    for (let defIndex = 0; defIndex < doc.definitions.length; defIndex++) {
+      const def = doc.definitions[defIndex];
+      if (def.kind !== 'ObjectTypeDefinition') {
+        continue;
+      }
+      if (['Query', 'Mutation', 'Subscription'].indexOf(def.name.value) > -1) {
+        let fields = def.fields.filter((field) => {
+          if (check(def.name.value, field.name.value, excludes)) {
             debug(`excluding ${def.name.value}.${field.name.value} from import`);
             return false;
           }
           return true;
         });
+
+        if (fields.length !== def.fields.length) {
+          mutations.push({
+            typeIndex,
+            defIndex,
+            fields
+          });
+        }
       }
     }
   }
 
+  for (const mutation of mutations) {
```

Comment:
it's a little more code, but this version avoids the deep copy and performs 10% - 15% slower than the original version of excludes